k8s/upgrades/0.9.0-1.0.0/jiva/jiva_volume_upgrade.sh

```diff
@@ -6,279 +6,298 @@
+    rm -r upgrade_tmp
 }
 trap 'on_exit' EXIT
 
-target_upgrade_version="1.0.0"
+upgrade_version="1.0.0"
 current_version="0.9.0"
 
+export upgrade_version
+export current_version
+
 function usage() {
     echo
     echo "Usage:"
     echo
-    echo "$0 <pv-name>"
+    echo "$0 <pv-name> <openebs-namespace>"
     echo
     echo "  <pv-name> Get the PV name using: kubectl get pv"
+    echo "  <openebs-namespace> Get the namespace where openebs control plane components are installed"
     exit 1
 }
 
-if [ "$#" -ne 1 ]; then
+function pre_check() {
+    local ns=$1
+    local pod_version=""
+    ## name=maya-apiserver label is common for both helm and operator yaml
+    maya_pod_name=$(kubectl get pod -n $ns \
+                 -l name=maya-apiserver    \
+                 -o jsonpath='{.items[0].metadata.name}' 2>&1)
+    rc=$?; 
+    if [ $rc -ne 0 ]; then 
+        reason=$(echo "$maya_pod_name" | tr --delete ":" )
+        patch_upgrade_task_error "PRE_UPGRADE" "failed to get maya apiserver pod name" "$reason" 
+        exit 1;
+    fi
+    pod_version=$(verify_openebs_version "pod" $maya_pod_name $ns 2>&1)
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        reason=$(echo "$pod_version" | tr --delete ":" )
+        patch_upgrade_task_error "PRE_UPGRADE" "failed to get maya apiserver pod name" "$reason"
+        exit 1
+    fi
+    if [ $pod_version != $upgrade_version ]; then
+        reason="expected version $upgrade_version but got $pod_version"
+        patch_upgrade_task_error "PRE_UPGRADE" "failed to verify OpenEBS control components" "$reason"
+        exit 1
+    fi
+}
+
+if [ "$#" -ne 2 ]; then
     usage
 fi
 
 pv=$1
+OpenEBS_ns=$2
+
+export pv
+
+pre_check $OpenEBS_ns
 
 # Check if pv exists
-kubectl get pv "$pv" &>/dev/null;check_pv=$?
+get_status=$(kubectl get pv "$pv" 2>&1);check_pv=$?
 if [ $check_pv -ne 0 ]; then
-    echo "$pv not found"
+    reason=$(echo $get_status | tr --delete ":")
+    patch_upgrade_task_error  "PRE_UPGRADE" "failed to get pv" "$reason"
     exit 1
 fi
 
 # Check if CASType is jiva
-cas_type=$(kubectl get pv "$pv" -o jsonpath="{.metadata.annotations.openebs\.io/cas-type}")
+cas_type=$(kubectl get pv "$pv" -o jsonpath="{.metadata.annotations.openebs\.io/cas-type}" 2>&1)
+rc=$?
+if [ $rc -ne 0 ]; then
+    reason=$(echo $cas_type | tr --delete ":")
+    patch_upgrade_task_error  "PRE_UPGRADE" "failed to get cas type for pv $pv" "$reason"
+    exit 1
+fi
 if [ "$cas_type" != "jiva" ]; then
-    echo "Jiva volume not found"
+    patch_upgrade_task_error  "PRE_UPGRADE" "failed to verify cas type for pv $pv" "invalid castype $cas_type"
     exit 1
-else [ "$cas_type" == "jiva" ];
-    echo "$pv is a jiva volume"
 fi
 
-ns=$(kubectl get pv "$pv" -o jsonpath="{.spec.claimRef.namespace}")
+#In case of DeployInOpenEBSNamespace is set to "true" check for pvc deployments in openebs namespace
+ns=$OpenEBS_ns
+count=$(kubectl get deploy -n "$ns" \
+        -l openebs.io/persistent-volume="$pv" | wc -l)
```

Comment:
space need to be trimmed as you discovered during your testing.. otherwise, count becomes 1 due to empty line in case of no deployments